pkg/kf/commands/apps/push.go

```diff
@@ -127,11 +129,28 @@ func NewPushCommand(p *config.KfParams, pusher kf.Pusher, b SrcImageBuilder) *co
 					}
 				}
 
+				var envMap map[string]string
+				var envVars []corev1.EnvVar
+				envVars, err = envutil.ParseCLIEnvVars(envs)
+				if err != nil {
+					return err
+				}
+				envMap = envutil.EnvVarsToMap(envVars)
+
+				// Merge cli envs over manifest envs
+				if app.Env == nil {
+					app.Env = envMap
```

Comment:
consider changing this to `make(map[string]string)` then always executing the else block to make straight-line reading this easier

Reply:
I thought about that too, but won't that not merge the variables then?

Reply:
I was thinking like this:

```
if app.Env == nil {
    app.Env = make(map[string]string)
}

for k, v := range envMap {
    app.Env[k] = v
}
```

Reply:
Ah, got it. I like that.

Reply:
This is done